python/tests/unit/connectors/ai/azure_ai_inference/services/test_azure_ai_inference_chat_completion.py

```diff
@@ -507,20 +509,31 @@ async def test_azure_ai_inference_streaming_chat_completion_with_function_choice
 
     mock_complete.return_value = mock_azure_ai_inference_streaming_chat_completion_response_with_tool_call
 
+    kernel.add_function(plugin_name="TestPlugin", function=decorated_native_function)
+
+    all_messages = []
     async for messages in azure_ai_inference_service.get_streaming_chat_message_contents(
         chat_history,
         settings,
         kernel=kernel,
         arguments=KernelArguments(),
     ):
-        assert len(messages) == 1
-        assert messages[0].role == "assistant"
-        assert messages[0].content == ""
-        assert messages[0].finish_reason == FinishReason.TOOL_CALLS
+        all_messages.extend(messages)
+
+    # Assert the number of total messages
+    assert len(all_messages) == 2, f"Expected 2 messages, got {len(all_messages)}"
+
+    # Validate the first message
+    assert all_messages[0].role == "assistant", f"Unexpected role for first message: {all_messages[0].role}"
+    assert all_messages[0].content == "", f"Unexpected content for first message: {all_messages[0].content}"
+    assert all_messages[0].finish_reason == FinishReason.TOOL_CALLS, (
```

Comment:
Why is the finish_reason of the first one `TOOL_CALLS`? The mock has the second message with the `TOOL_CALLS` as the finish reason.

Reply:
The first finish reason is related to the FunctionCallContent.

Reply:
Oh, I see it has an empty list of choices thus the first message was skipped.